Purchases/FoundationExtensions/Error+Extensions.swift

```diff
@@ -15,6 +15,10 @@ import Foundation
 
 extension Error {
 
+    var isAPIBlockedError: Bool {
+        return (self as NSError).userInfo[DNSChecker.blockedErrorKey] as? Bool ?? false
```

Comment:
Should we check the domain and code here instead? Seems weird to check if an error is a certain type just by looking at the contents of `userInfo`.

Reply:
it is kinda weird, but part of the logic in determining if it is a DNS blocked error is that the DNSChecker does DNS resolution. Having the domain and code exist isn't enough to guarantee it's an APIBlockedError. We need to attempt to resolve the DNS and get specific values back to be sure. At that point, I set the `[blockedErrorKey] = true`.

Reply:
>  Having the domain and code exist isn't enough to guarantee it's an APIBlockedError.

But we can make it guaranteed if we create a specific error code for api blocked errors, right?